### ISSUE TITLE
Can't test flash messages with inertia views

### ISSUE THREAD

**Issue by markstory** [ishanvyas22/cakephp-inertiajs#17]
**Describe the bug**
I want to write test assertions for flash messages being set inside of controllers that render inertia views. An example test would be:

```php
    /**
     * Test edit method
     *
     * @return void
     */
    public function testEdit(): void
    {
        $this->login();
        $this->enableCsrfToken();
        $this->enableRetainFlashMessages();

        $this->post('/users/profile', [
            'email' => 'badthings@example.com',
            'unverified_email' => 'example@example.com',
            'timezone' => 'America/San_Francisco',
        ]);
        $this->assertRedirect('/todos/today');
        $this->assertFlashElement('flash/success');
    }
```

However flash messages are never retained and the assertion always fails. I've tracked this down to `InertiaResponseTrait` deleting flash messages *before*  the test harness can capture them via the `View.beforeRender` event.

## Proposal

My proposal to fix this is to move the flash message and CSRF view variable setting to happen inside the View `render()` methods instead of using the `Controller.beforeRender` event.

If this sounds OK with you, I'm happy to put a pull request together for it.

**Comment by markstory** [comment]
I took a further look at this and only moving the flash data manipulation to the View won't entirely fix the problem as the session data will still be deleted before the `View.beforeRender` event is triggered. I think this might need to be fixed in `IntegrationTestTrait` instead.

**Comment by ishanvyas22** [comment]
Hi @markstory, I think you need to change the method of testing here as we are not working with CakePHP's elements when working with Inertia instead we have Vue, React, etc. components to render the flash message. So this assertion won't work here.
I suggest you assert data from `flash` key(which contains flash message, element name, params, etc.) in the returned response which will be in json when working with Inertia responses.

This is the only way I can think right now to assert this, let me know if anyone have better idea. Any suggestion would be welcomed.

**Comment by ishanvyas22** [comment]
Seems like you've already fixed this with https://github.com/cakephp/cakephp/pull/15262 PR. Is everything working fine after merge that in? Also do we need some docs on this?

**Comment by markstory** [comment]
Doing assertions on the view variables was another approach I considered. I thought that the flash assertions were better as they would keep tests more consistent between inertia and non-inertia based controllers.

The changes in CakePHP that I made should resolve this problem. I started off thinking of fixing this plugin, but came to the conclusion that there could be other view implementations that work in a similar way.
